tester/constraint_test.go

```diff
@@ -0,0 +1,182 @@
+	s.Assert().Nil(c)
+	s.Assert().EqualError(err, tester.ErrInvalidAggregator.Error())
+	// Valid constraint - AVG.
+	c, err = tester.ParseConstraint("AVG(errors) < 10", s.parsers...)
+	s.Assert().NotNil(c)
+	s.Assert().NoError(err)
+	// Valid constraint - MIN.
+	c, err = tester.ParseConstraint("MIN(errors) < 5", s.parsers...)
+	s.Assert().NotNil(c)
+	s.Assert().NoError(err)
+	// Valid constraint - MAX.
+	c, err = tester.ParseConstraint("MAX(errors) < 20", s.parsers...)
+	s.Assert().NotNil(c)
+	s.Assert().NoError(err)
+	// Invalid metric '0xdeadbeef'.
+	c, err = tester.ParseConstraint("MAX(0xdeadbeef) < 10", s.parsers...)
+	s.Assert().Nil(c)
+	s.Assert().EqualError(err, tester.ErrNotParsed.Error())
+	// Should return error if metric parser returned error.
+	c, err = tester.ParseConstraint("MAX(mymetrics) > 1", []tester.MetricParser{
+		func(_ string) (tester.Metric, error) {
+			return nil, errors.New("error")
+		},
+	}...)
+	s.Assert().Nil(c)
+	s.Assert().Error(err)
+	// Valid metric - errors.
+	c, err = tester.ParseConstraint("MAX(errors) < 10", s.parsers...)
+	s.Assert().NotNil(c)
+	s.Assert().NoError(err)
+	// Valid metric - latency.
+	c, err = tester.ParseConstraint("MAX(latency) < 100", s.parsers...)
+	s.Assert().NotNil(c)
+	s.Assert().NoError(err)
+	// Invalid comparator '@@'.
+	c, err = tester.ParseConstraint("MAX(errors) @@ 10", s.parsers...)
+	s.Assert().Nil(c)
+	s.Assert().EqualError(err, tester.ErrInvalidComparator.Error())
+	// Valid threshold.
+	c, err = tester.ParseConstraint("MIN(latency) < 1.337", s.parsers...)
+	s.Assert().NotNil(c)
+	s.Assert().NoError(err)
+	// Invalid threshold - comma.
+	c, err = tester.ParseConstraint("MAX(latency) < 1,337", s.parsers...)
+	s.Assert().Nil(c)
+	s.Assert().Error(err)
+	// Invalid threshold - hex.
+	c, err = tester.ParseConstraint("MAX(latency) < 0x414141", s.parsers...)
+	s.Assert().Nil(c)
+	s.Assert().Error(err)
+	// Invalid threshold - overflow.
+	number := `17976931348623157081452742373170435679807056752584499659891747680
+31572607800285387605895586327668781715404589535143824642343213268894641827684675
+46703537516986049910576551282076245490090389328944075868508455133942304583236903
+22294816580855933212334827479782620414472316873817718091929988125040402618412485
+83681337`
+	c, err = tester.ParseConstraint(
+		fmt.Sprintf(
+			"MAX(latency) < %s",
+			strings.Replace(number, "\n", "", -1),
+		),
+		s.parsers...,
+	)
+	s.Assert().Nil(c)
+	s.Assert().Contains(err.Error(), "value out of range")
+}
+
+func (s *ParseConstraintTestSuite) TestCheck() {
+	// No datapoints should result in error.
+	s.metric.On("Fetch", mock.Anything, mock.Anything).
+		Return([]tester.DataPoint(nil), nil).Once()
+	c := &tester.Constraint{Metric: s.metric}
+	err := c.Check(time.Now(), time.Second)
+	s.Assert().Error(err)
+	s.metric.AssertExpectations(s.T())
+	// If metric.Fetch returned error, Check should result in error.
+	s.metric.On("Fetch", mock.Anything, mock.Anything).
```

Comment:
When we know actual parameters of the call it's better to explicitly pass them instead of relying on `mock.Anything`